HR_allfuncs.py

```diff
@@ -118,149 +122,155 @@ def HR_peakdetect(data_array):
     return time_array
 
 
-def instHR(t_array):
-    """
-            .. function:: instHR(t_array)
-            Calculates instant heart rate from peak time array
-            :param t_array: a 1-d list with time values of peaks
-    """
-    dt_first_beat = t_array[2] - t_array[1]
 
-    instHR = 1/dt_first_beat * 60
 
-    return instHR
 
+class Vitals:
 
 
+    def __init__(self):
+        self.avg_hr_val = float
+        self.inst_hr_val = float
 
+    def hr_averaging(self, averaging_time, time_array):
+        num_arg = 3
+        min_to_sec = 60
 
-def hr_averaging(averaging_time,time_array):
 
-    """
-    .. function:: hr_averaging(averaging_time, tachy_limit = 100, brachy_limit = 60)
+        """
+        .. function:: hr_averaging(averaging_time, tachy_limit = 100, brachy_limit = 60)
 
-    Calculate the average HR based upon ECG data.
+        Calculate the average HR based upon ECG data.
 
-    :param averaging_time: time period (in min) used to calculate average HR
-    :param time_array: the time_array after peak_detect has been called
-    :param brachy_limit: sets the brachycardia threshold limit, defaults to 100
-    :rtype: integer value of average HR
-    """
+        :param averaging_time: time period (in min) used to calculate average HR
+        :param time_array: the time_array after peak_detect has been called
+        :rtype: integer value of average HR
+        """
 
-    # check for valid num_args
-    try:
-        num_arg == sys.argv
-    except TypeError:
-        print("Please input the correct number of arguments")
-    try:
-        test_fraction = Fraction(averaging_time)
-        averaging_time = float(Fraction(test_fraction))
-    except ValueError:
-        print("That is not a valid fraction, float, or int")
-        raise ValueError
-    except ZeroDivisionError:
-        print("You cannot divide by zero")
-        raise ZeroDivisionError
 
-    # check for non-zero averaging time
-    if averaging_time <= 0:
-        print("Your averaging time input must be greater than zero.")
-        raise ValueError
 
-    # convert averaging time to float
-    try:
-        averaging_time = float(averaging_time)
-    except TypeError:
-        print("Your averaging time input is not a a valid number, please input a number.")
+        # check for valid num_args
+        try:
+            num_arg == sys.argv
```

Comment:
hmmm so this is checking to see if the right number of command line args were passed into the program? This should probably be tested in the main function as opposed to the class...we can chat more about this is needed

Reply:
@yashasaxena

Reply:
Yup, that's the plan! Just didn't get a chance to modify this for the update